### PR TITLE
Adds page about recovering from a failed job to anomaly detection docs

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds a new page to the anomaly detection documentation that explains how to recover from a failed job.

### Preview

[Recover from a failed anomaly detection job](https://stack-docs_1667.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-recover-failed-jobs.html)

### CODE REVIEW

File: docs/en/stack/ml/anomaly-detection/ml-configuration.asciidoc [hunk 1]
```diff
@@ -18,9 +18,11 @@ you visualize and explore the results.
 
 * <<create-jobs>>
 * <<stopping-ml>>
+* <<ml-recover-failed-jobs>>
 
-After you learn how to create and stop {anomaly-detect} jobs, you can check the 
-<<anomaly-examples>> for more advanced settings and scenarios.
+After you learn how to create and stop {anomaly-detect} jobs, and how to recover 
+from a failed job, you can check the <<anomaly-examples>> for more advanced 
+settings and scenarios.
```
Comment (droberts195) [inline docs/en/stack/ml/anomaly-detection/ml-configuration.asciidoc:25]: I am not sure about mentioning recovering from failed jobs in this sentence.  It makes it sound like it's one of the first things you have to learn when it's not really.  It's just something that needs to be easy to find out about _if_ you do need it.

**Review by droberts195 — APPROVED** [review]
LGTM

File: docs/en/stack/ml/anomaly-detection/ml-recover-failed-jobs.asciidoc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+If a persistent problem caused the job to fail, find out which node the failed 
+job was running on by checking the job stats on the **Job management** pane in 
+{kib} before the recovery procedure. Then get the logs for that node and look 
+for exceptions and errors where the ID of the {anomaly-job} is in the message to 
+have a better understanding of the issue.
+
+If an {anomaly-job} has failed, do the following to recover from `failed` state: 
+
+. _Force_ stop the corresponding {dfeed} by using the
```
[the remark this answers is not in the document]
Comment (szabosteve) [reply docs/en/stack/ml/anomaly-detection/ml-recover-failed-jobs.asciidoc:18]: The only place where I saw the force close option on the UI is when you want to revert to a previous snapshot. I checked it with the UI folks and it seems they don't set a force flag on the UI.

[the remark this answers is not in the document]
Comment (lcawl) [reply docs/en/stack/ml/anomaly-detection/ml-recover-failed-jobs.asciidoc]: Okay cool.  I saw it mentioned in https://github.com/elastic/kibana/pull/20927, but that might be for a very specific error code only.

**Comment by szabosteve** [comment]
@lcawl I addressed your feedback via ddfd287

**Pull request merged**